cloudevents-sdk-mqtt/Cargo.toml

```
@@ -0,0 +1,18 @@
+[package]
+name = "cloudevents-sdk-mqtt"
+version = "0.2.0"
+authors = ["Francesco Guardiani <francescoguard@gmail.com>"]
+license-file = "../LICENSE"
+edition = "2018"
+description = "CloudEvents official Rust SDK - Mqtt integration"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+cloudevents-sdk = { version = "0.2.0", path = ".." }
+lazy_static = "1.4.0"
+paho-mqtt = { path = "../../paho.mqtt.rust" }
```

Why are you adding the dependency by path and not by cargo by default dependency?

```suggestion
paho-mqtt = {version="0.7.1"}
```

---

@Lazzaretti , the version `0.7.1` has an issue. It only allows property from head of container.

https://github.com/eclipse/paho.mqtt.rust/blob/7b3a29cdca902c3b9f0ab20909ed24bd47b6e881/src/properties.rs#L466

this is fixed in current master

https://github.com/eclipse/paho.mqtt.rust/blob/master/src/properties.rs#L900

Unfortunately, there is no release for it yet. 
According to cloudevents specification, we have to put all headers under prop code `UserProperty`.

https://github.com/cloudevents/spec/blob/master/mqtt-protocol-binding.md#314-examples

---

Can we get a new release of paho.mqtt soon?

---

@slinkydeveloper @Lazzaretti i currently linked it to git repo.

---

Still, this is not valid to perform a release of the package to creates.io

---

In any case, we need a release from them. @fpagliughi Are you going to perform a release of paho.mqtt any time soon?

---

Yes, apologies for the delay. Lots going on these days. I'm starting work back on Paho Rust this week. I think the strategy will be to push out what's been sitting up on GitHub for the last few monhts as v0.8, so as not to break the code people have been creating with it. That should happen before the end of the week.

Then work will start to update the underlying C lib and add some new features. That should go out by the end of the year as v0.9, and will have some minor breaking changes. So people can choose.